### PR TITLE
Amend code to make it easier for user to change PI/Co-I on a proposal

### PR DESCRIPTION
**Pull request by RasmiaKulan**
Closes: [788](https://github.com/UserOfficeProject/user-office-project-issue-tracker/issues/788)
## Description

Made changes to make it easier for users to switch between PI's and Co-I's on a proposal
## Motivation and Context
Users can now switch between the Principal Investigator's and Co-Investigator's in the Co-Proposers table with a click of a button.
tempsnip

## Fixes
[788](https://github.com/UserOfficeProject/user-office-project-issue-tracker/issues/788)

## Changes

Files Changed:
                       user-office-frontend/src/components/proposal/ParticipantModal.tsx
                       user-office-frontend/src/components/proposal/ProposalParticipant.tsx
                       user-office-frontend/src/components/proposal/ProposalParticipants.tsx
                       user-office-frontend/src/components/proposalBooking/CreateUpdateVisit.tsx
                       user-office-frontend/src/components/questionary/questionaryComponents/ProposalBasis/QuestionaryComponentProposalBasis.tsx
                       apps/user-office-frontend/src/components/user/PeopleTable.tsx
## Tests included/Docs Updated?

- [ ] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: apps/user-office-frontend/src/components/proposal/ProposalParticipant.tsx
Comment (mutambaraf) [inline apps/user-office-frontend/src/components/proposal/ProposalParticipant.tsx:27]: If it can be null or undefined should not we make it optional the handle that state in the component. We can also be more descriptive with the prop name form pi to principalInvestigator.

File: apps/user-office-frontend/src/components/proposal/ParticipantModal.tsx
Comment (mutambaraf) [inline apps/user-office-frontend/src/components/proposal/ParticipantModal.tsx:55]: It would be nice to be more descriptive with the setPi to setPrincipalInvestigator

File: apps/user-office-frontend/src/components/proposal/ParticipantModal.tsx
Comment (mutambaraf) [inline apps/user-office-frontend/src/components/proposal/ParticipantModal.tsx:33]: This can also be updated to setPrincipalInvestigator

File: apps/user-office-frontend/src/components/proposal/ProposalParticipants.tsx
Comment (mutambaraf) [inline apps/user-office-frontend/src/components/proposal/ProposalParticipants.tsx:34]: We may not need this comment.

File: apps/user-office-frontend/src/components/proposal/ProposalParticipants.tsx
Comment (mutambaraf) [inline apps/user-office-frontend/src/components/proposal/ProposalParticipants.tsx:45]: We may not need this comment.

File: apps/user-office-frontend/src/components/proposalBooking/CreateUpdateVisit.tsx
Comment (mutambaraf) [inline apps/user-office-frontend/src/components/proposalBooking/CreateUpdateVisit.tsx:92]: This can be made optional if it not needed by other components we will not need to pass it to them.

File: apps/user-office-frontend/src/components/questionary/questionaryComponents/ProposalBasis/QuestionaryComponentProposalBasis.tsx
Comment (mutambaraf) [inline apps/user-office-frontend/src/components/questionary/questionaryComponents/ProposalBasis/QuestionaryComponentProposalBasis.tsx:60]: We can make this more descriptive.

File: apps/user-office-frontend/src/components/questionary/questionaryComponents/ProposalBasis/QuestionaryComponentProposalBasis.tsx
Comment (mutambaraf) [inline apps/user-office-frontend/src/components/questionary/questionaryComponents/ProposalBasis/QuestionaryComponentProposalBasis.tsx:143]: Lets be more descriptive here too.

File: apps/user-office-frontend/src/components/user/PeopleTable.tsx
Comment (mutambaraf) [inline apps/user-office-frontend/src/components/user/PeopleTable.tsx:272]: Why are we using link button as an icon here. We can also use CSS to style the inline text. Just verify if we are not potential wrapping a button inside another button

**Review by mutambaraf — CHANGES_REQUESTED** [review]
Another minor change we can fix here.

**Comment by mutambaraf** [comment]
Is there any reason we closed https://github.com/UserOfficeProject/user-office-core/pull/232

**Comment by RasmiaKulan** [comment]
> Is there any reason we closed #232

It was closed as it had clashing merge issues. 🙁

**Comment by Scott-James-Hurley** [comment]
Might it be an idea to add a couple of e2e tests for this new functionality?

**Comment by RasmiaKulan** [comment]
This issue is being blocked by a [bug](https://github.com/UserOfficeProject/user-office-project-issue-tracker/issues/856) that is causing the e2e test to fail.

**Comment by mutambaraf** [comment]
> This issue is being blocked by a [bug](https://github.com/UserOfficeProject/user-office-project-issue-tracker/issues/856) that is causing the e2e test to fail.

I this for the e2e or it the the feature which requires the changes.

**Comment by RasmiaKulan** [comment]
> > This issue is being blocked by a [bug](https://github.com/UserOfficeProject/user-office-project-issue-tracker/issues/856) that is causing the e2e test to fail.
> 
> I this for the e2e or it the the feature which requires the changes.

Its a current bug in the system. We tested it and its on prod and dev as well

**Comment by simonfernandes** [comment]
@UserOfficeProject/swap 

The last problem with this is that when using the new button to quick-switch from Co-I to PI, an empty name displays in the PI input. It's because the basic user details can't be fetched due to lack of permissions. I did a fix for STFC but didn't realise it affected the ESS too: https://github.com/UserOfficeProject/user-office-core/pull/261.

For the ESS, since you don't have a concept of searchable users (anybody shows up when searching by email I believe), could we change the getBasic function to be allowed for any user (i.e. even when there is no relation to the proposal)? It was originally added in https://github.com/UserOfficeProject/user-office-backend/pull/480/commits/bf020c6db745a206d42fc0b79132a031021cff62 for GDPR.

**Comment by martin-trajanovski** [comment]
> 

@simonfernandes I think this is fine to some extent. Because it will show a user only if you fully match their email. But I am afraid that it is not OK for any logged in user to be able to hit the endpoint and query the user details just like this without any restrictions. Maybe @bolmsten (or optionally @enthusiastio) can answer this a bit better but he is coming back next week I am not sure how long you can wait with the PR.

**Comment by simonfernandes** [comment]
@bolmsten @enthusiastio 

Do you have any thoughts on whether this can be done?

**Comment by bolmsten** [comment]
> @bolmsten @enthusiastio
> 
> Do you have any thoughts on whether this can be done?

I do not understand why we would need to query the basicUserData endpoint, when switching a PI and Co do we not have all necessary information already in the frontend?

**Comment by enthusiastio** [comment]
> @bolmsten It's to populate the user details of the PI field. The PI is set to the user ID but the hook that gets and displays the user details queries the basicUserData endpoint which requires the user to have a previous affiliation to the proposal. This is actually a bug in your version of the software even without this feature - you can test by searching for a user who you've had no prior involvement with (this queries a different user details by email endpoint), and then you'll see the PI field is blank.

Hi, sorry for the late response. I can't seem to reproduce it has this been addressed
Peek 2023-08-27 19-07

File: apps/user-office-frontend/src/components/user/PeopleTable.tsx
Comment (enthusiastio) [inline apps/user-office-frontend/src/components/user/PeopleTable.tsx:296]: This line produces an error in the console that the button can not be a child of a button. Maybe try this instead
```suggestion
          onClick={(event) => event.preventDefault()}
```

**Review by enthusiastio — APPROVED** [review]
Check the small comment before merge

**Comment by bolmsten** [comment]
> > @bolmsten It's to populate the user details of the PI field. The PI is set to the user ID but the hook that gets and displays the user details queries the basicUserData endpoint which requires the user to have a previous affiliation to the proposal. This is actually a bug in your version of the software even without this feature - you can test by searching for a user who you've had no prior involvement with (this queries a different user details by email endpoint), and then you'll see the PI field is blank.
> 
> Hi, sorry for the late response. I can't seem to reproduce it has this been addressed Peek 2023-08-27 19-07

I can reproduce it now

**Comment by bolmsten** [comment]
I think this PR: https://github.com/UserOfficeProject/user-office-core/pull/314 should fix the issue. 

@simonfernandes  @enthusiastio

**Pull request closed without merging**
